tests/Doctrine/Tests/ORM/Tools/AttachEntityListenersListenerTest.php

```diff
@@ -56,6 +56,17 @@ public function testAttachEntityListeners(): void
         self::assertCount(1, $metadata->entityListeners['postLoad']);
         self::assertEquals('postLoadHandler', $metadata->entityListeners['postLoad'][0]['method']);
         self::assertEquals(AttachEntityListenersListenerTestListener::class, $metadata->entityListeners['postLoad'][0]['class']);
+
+        // Can reattach entity listeners even class metadata factory recreated.
+        $factory2 = new ClassMetadataFactory();
+        $factory2->setEntityManager($this->em);
+
+        $metadata2 = $factory2->getMetadataFor(AttachEntityListenersListenerTestFooEntity::class);
+
+        self::assertArrayHasKey('postLoad', $metadata2->entityListeners);
+        self::assertEquals(AttachEntityListenersListenerTestListener::class, $metadata2->entityListeners['postLoad'][0]['class']);
+        self::assertCount(1, $metadata2->entityListeners['postLoad']);
```

Comment:
Please switch those 2 lines with each other. Checking that `$metadata2->entityListeners['postLoad']` has elements should happen before the check where actual access to that element happens.

Reply:
Oh my mistake, switched, thank you!